sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureBasicFileAttributes.java

```
@@ -0,0 +1,142 @@
+    latter introduces virtual directories into the actual code path/api surface. While we are clear in our docs about
+    the possible pitfalls of virtual directories, and customers should be aware of it, they shouldn't have to code
+    against it. Therefore, we fall back to documenting that reading attributes on a virtual directory will throw. And
+    who reads attributes on a directory anyway?
+     */
+    AzureBasicFileAttributes(Path path) throws IOException {
+        try {
+            this.properties = new AzureResource(path).getBlobClient().getProperties();
+        } catch (BlobStorageException e) {
+            throw LoggingUtility.logError(logger, new IOException(e));
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public FileTime lastModifiedTime() {
+        return FileTime.from(properties.getLastModified().toInstant());
+    }
+
+    /**
+     * Unsupported.
+     * {@inheritDoc}
+     */
+    @Override
+    public FileTime lastAccessTime() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public FileTime creationTime() {
+        return FileTime.from(properties.getCreationTime().toInstant());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isRegularFile() {
+        return !this.properties.getMetadata().getOrDefault(AzureResource.DIR_METADATA_MARKER, "false").equals("true");
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Will only return true if the directory is a concrete directory. See
+     * {@link AzureFileSystemProvider#createDirectory(Path, FileAttribute[])} for more information on virtual and
+     * concrete directories.
+     */
+    @Override
+    public boolean isDirectory() {
+        return !this.isRegularFile();
+    }
+
+    /**
+     * @return false. Symbolic links are not supported.
+     */
+    @Override
+    public boolean isSymbolicLink() {
+        return false;
```

Should this throw `UnsupportedOperationException` if it isn't supported?

---

I'm not sure. In some sense it's just always false, right? But maybe it makes it more clear that sym links aren't supported if I throw?

---

I think returning always `false` is correct. Throwing `UnsupportedOperationException` would mean "I cannot check if file is symlink or not` in plain English. I believe our file system doesn't have symlinks so all files are real by definition, therefore we know how to check if they're symlinks or not.